include/xtensor/xsort.hpp

```diff
@@ -144,6 +144,163 @@ namespace xt
         return sort(de, de.dimension() - 1);
     }
 
+    namespace detail {
+        template <class VT, class T>
+        struct rebind_value_type
+        {
+            using type = xarray<VT, xt::layout_type::dynamic>;
+        };
+
+        template <class VT, class EC, layout_type L>
+        struct rebind_value_type<VT, xarray<EC, L>>
+        {
+            using type = xarray<VT, L>;
+        };
+
+        template <class VT, class EC, size_t N, layout_type L>
+        struct rebind_value_type<VT, xtensor<EC, N, L>>
+        {
+            using type = xtensor<VT, N, L>;
+        };
+
+        template <class VT, class ET, class S, layout_type L>
+        struct rebind_value_type<VT, xtensor_fixed<ET, S, L>>
+        {
+            using type = xtensor_fixed<VT, S, L>;
```

Comment:
This should probably be a `xtensor_fixed` of dimension 1, and length `detail::fixed_compute_size<S>::value`.

Reply:
at least for the case where the second argument is xnone (or xtuph)

Reply:
here is a usage example for the same thing https://github.com/QuantStack/xtensor/pull/922/files#diff-80059cad53fa3b496f127a18946f1f0fR133 (note that PR has not been merged yet).